src/metricscollector/cmd/metricscollector/main.go

```diff
@@ -67,15 +75,33 @@ func main() {
 	noaa := consumer.New(dopplerUrl, tlsConfig, nil)
 	noaa.RefreshTokenFrom(cfClient)
 
-	var database db.DB
-	database, err = sqldb.NewSQLDB(&conf.Db, logger.Session("db"))
+	var metricsDB db.MetricsDB
+	metricsDB, err = sqldb.NewMetricsSQLDB(conf.Db.MetricsDbUrl, logger.Session("metrics-db"))
+	if err != nil {
+		logger.Error("failed to connect metrics database", err, lager.Data{"url": conf.Db.MetricsDbUrl})
+		os.Exit(1)
+	}
+	defer metricsDB.Close()
+
+	var policyDB db.PolicyDB
+	policyDB, err = sqldb.NewPolicySQLDB(conf.Db.PolicyDbUrl, logger.Session("policy-db"))
 	if err != nil {
-		logger.Error("failed to connect database", err, lager.Data{"conf": conf.Db})
+		logger.Error("failed to connect policy database", err, lager.Data{"url": conf.Db.PolicyDbUrl})
 		os.Exit(1)
 	}
-	defer database.Close()
+	defer policyDB.Close()
+
+	mcClock := clock.NewClock()
+	createPoller := func(appId string) collector.AppPoller {
+		return collector.NewAppPoller(appId, conf.Collector.PollInterval, logger.Session("app-poller"), cfClient, noaa, metricsDB, mcClock)
+	}
+
+	collectServer := collector.NewCollector(conf.Collector.RefreshInterval, logger.Session("collector"), policyDB, mcClock, createPoller)
+	collectServer.Start()
```

Comment:
@fraenkel 
do you mean we remove the start / stop function of collector, and get all the logic in  Run function ?  instead of just having a new Run func to wrap the existing Start/Stop func